packages/itwin/property-grid/src/property-grid-react/PropertyGridUiItemsProvider.tsx

```diff
@@ -113,34 +116,56 @@ export type PropertyGridWidgetProps = PropertyGridComponentProps & {
    */
   widgetId?: string;
 } & (
-    | {
-        /**
-         * Predicate indicating if the widget should be shown for the current selection set.
-         * @deprecated in 1.16. Use the overload taking `Selectables` instead.
-         */
-        shouldShow?: (selection: Readonly<KeySet>) => boolean;
-        selectionStorage?: never;
-      }
-    | {
-        /** Predicate indicating if the widget should be shown for the current selection set. */
-        shouldShow?: (selection: Selectables) => Promise<boolean>;
-
-        /**
-         * Unified selection storage to use for listening and getting active selection.
-         *
-         * When not specified, the deprecated `SelectionManager` from `@itwin/presentation-frontend` package
-         * is used.
-         */
-        selectionStorage: SelectionStorage;
-      }
-  );
+  | {
+      /**
+       * Predicate indicating if the widget should be shown for the current selection set.
+       * @deprecated in 1.16. Use the overload taking `Selectables` instead.
+       */
+      shouldShow?: (selection: Readonly<KeySet>) => boolean;
+      selectionStorage?: never;
+    }
+  | {
+      /** Predicate indicating if the widget should be shown for the current selection set. */
+      shouldShow?: (selection: Selectables) => Promise<boolean>;
+
+      /**
+       * Unified selection storage to use for listening and getting active selection.
+       *
+       * When not specified, the deprecated `SelectionManager` from `@itwin/presentation-frontend` package
+       * is used.
+       */
+      selectionStorage: SelectionStorage;
+    }
+);
 
-/** Component that renders `PropertyGridComponent` an hides/shows widget based on `UnifiedSelection`. */
-// eslint-disable-next-line deprecation/deprecation
-function PropertyGridWidget({ shouldShow, widgetId, ...props }: PropertyGridWidgetProps) {
+/**
+ * Props for `createPropertyGrid`.
+ * @public
+ */
+export type PropertyGridWidgetProps = PropertyGridComponentProps & PropertyGridWidgetOwnProps;
+
+/** @internal */
+type PropertyWidgetInternalProps = PropertyGridWidgetOwnProps & {
+  PropertyGridComponent: ReactNode;
```

Comment:
We should be consistent with property names.
```suggestion
  propertyGridComponent: ReactNode;
```

I think it should be in PascalCase only if you take it as `React.ComponentType<SomeProps>` and use it like
```tsx
interface Props {
    OtherComponent: React.ComponentType<SomeProps>
}

function MyComponent({ OtherComponent }: Props) {
    return <div>
        <OtherComponent {...someProps} />
    </div>
}
```

Reply:
changed